front/lib/api/assistant/actions/retrieval.ts

```diff
@@ -148,6 +150,71 @@ export function renderRetrievalActionForModel(
     content,
   };
 }
+export function renderAssistantFunctionCallMessageForRetrievalAction(
+  action: RetrievalActionType
+): AssistantFunctionCallMessageTypeModel {
+  const timeFrame = action.params.relativeTimeFrame;
+  const params = {
+    query: action.params.query,
+    relativeTimeFrame: timeFrame
+      ? `${timeFrame.duration}${timeFrame.unit}`
+      : "all",
+    topK: action.params.topK,
+  };
+
+  return {
+    role: "assistant" as const,
+    content: null,
+    function_calls: [
+      {
```

Comment:
only the function_call should be returned here. Methods need to change name

Reply:
I added a commit 🙏🏻